tests/src/Kernel/Controller/WopiControllerTest.php

```diff
@@ -161,29 +211,58 @@ protected function doTestWopiPutFile(
       content: $new_file_content,
     );
     $request->headers->add($request_headers);
+
+    $request_time = \Drupal::time()->getRequestTime();
     $this->assertJsonResponseOk(
       [
-        'LastModifiedTime' => $file_changed_time->format('c'),
+        'LastModifiedTime' => DateTimeHelper::format($request_time),
       ],
       $request,
     );
+
     $media = Media::load($this->media->id());
-    $this->assertSame($reason_message, $media->getRevisionLogMessage());
+    $file = $this->loadCurrentMediaFile();
+
+    // File entity and content are updated.
+    $this->assertSame($request_time, $file->getChangedTime());
+    $actual_file_content = file_get_contents($file->getFileUri());
+    $this->assertSame($new_file_content, $actual_file_content);
+    $this->assertSame(strlen($new_file_content), $file->getSize());
+
+    if (!$new_file) {
+      // The URI remains the same, and no File entity has been created.
+      $this->assertSame($old_file->id(), $file->id());
+      $this->assertSame($file->getFileUri(), $file->getFileUri());
```

Comment:
```suggestion
      $this->assertSame($old_file->getFileUri(), $file->getFileUri());
```
But I would be careful when using `$old_file` here.
In the previous version of the test we could assume that `$old_file` was not changed, because a new file entity was created. But now, what if the `$old_file` and `$file` are actually the same object?
If we want to be sure, we have to read the values from `$old_file` _before_ we do the save request, keep them in variables, and then compare after the save.
Also, I think we can do all of the same checks here like `->isPermanent()`, `->getOwnerId()`. Possibly as shared piece of code, outside the if().

Reply:
perhaps `clone $old_file` would also do the trick, but I am not sure. Local variables are the most reliable, but also more verbose.

Reply:
I prefer local variables too.
About the other checks, didn't want to do checks about things we do not change.

Reply:
So we should have something like this:
(where `$old_file_uri` needs to be set before the operation)
```suggestion
      $this->assertSame($old_file_uri, $file->getFileUri());
```

Reply:
> About the other checks, didn't want to do checks about things we do not change.

ok.
I was thinking of the next time we refactor this and the two cases share part of the execution path.
But we can update the test when that happens.

Reply:
My bad about the comparisson, changed